manifests/sar_metric.pp

```diff
@@ -33,27 +42,72 @@
                             ' > /dev/null',
                             ], '')
 
-  cron { "${metrics_type}_metrics_collection" :
-    ensure  => $metric_ensure,
-    command => $metrics_command,
-    user    => 'root',
-    minute  => $cron_minute,
-  }
-
   # The hardcoded numbers with the fqdn_rand calls are to trigger the metrics_tidy 
   # command to run at a randomly selected time between 12:00 AM and 3:00 AM.
   # NOTE - if adding a new service, the name of the service must be added to the valid_paths array in files/metrics_tidy
 
-  cron { "${metrics_type}_metrics_tidy" :
+  $tidy_command = "${puppet_metrics_collector::scripts_dir}/metrics_tidy -d ${metrics_output_dir} -r ${retention_days}"
+
+  file {"/etc/systemd/system/${metrics_type}-metrics.service":
```

Comment:
Nit: This same pattern is used across many classes. It would be better to create a defined type that creates the timer and services and just add that resource in these classes.

Reply:
I agree, but this may be better left for a refactor type of ticket.  While making these changes I saw a lot of opportunities for refactoring, but decided to leave that them for later.

Reply:
We went from 2 cron resources that were duplicated across these classes to 4 file resources and 4 service resources. It seems like that increase in duplicate code went from 13 lines to 46 lines. The extra 33 lines in each manifest cross a threshold that would be worthwhile to do in this change.

Reply:
Now that I understand the scope of this ask, it should be fixed in 06ad7dcc.  This resulted in no changes on my test box and the tests are still passing.

Reply:
Looks great. Thanks for doing that.